### PR TITLE
Check for the ref value to show the onReset prop

### PR DESCRIPTION
**Pull request by glenngijsberts**
# Description

Instead of having the logic of showing outside the component, this PR will make it possible to have the logic to show the onReset prop inside the component. See the Storybook story to see how it works without the previous code.

### CODE REVIEW

File: src/components/Input/index.js [hunk 1]
```diff
@@ -214,7 +214,14 @@ export const Input = React.forwardRef(
...
       (props.value && props.value.length && props.onReset) ||
-      (!props.value && ref && props.onReset)
+      (!props.value &&
+        ref &&
+        props.onReset &&
+        ((ref &&
+          ref.current &&
+          ref.current.value &&
+          ref.current.value.length) ||
+          false))
```
Comment (rppld) [inline src/components/Input/index.js]: Bit complex now this part. Maybe add a short comment explaining what’s happening? Controlled with value, uncontrolled with ref and value, etc.

Comment (RobVermeer) [reply src/components/Input/index.js]: Force pushed a change, let me know what you think.

**Review by RobVermeer — APPROVED** [review]
👌

**Review by rppld — APPROVED** [review]

**Pull request merged**